dissect/target/plugins/os/windows/wer.py

```diff
@@ -34,13 +48,23 @@ def _collect_wer_data(wer_file: Path) -> tuple[list[tuple[str, str]], dict[str,
                 category, name = name.split(".", 1)
                 key = f"{category.split('[')[0]}{name}"
 
+            # array-like entries
+            elif arraylike_pattern.match(name):
+                match = arraylike_pattern.match(name)
+                key = match.group(1) + "_" + match.group(2)
+
             if "EventTime" in name:
                 value = wintimestamp(int(value))
                 record_type = "datetime"
                 key = "ts"
 
             key = _key_to_snake_case(key if key else name)
 
+            # Windows wer files might contain any valid unicode character in its field names
+            # but we use the field name as python identifier. So, some character are not allowed
+            # to be used.
+            key = _clean_key(key)
```

Comment:
I had a windows10 vm that had keys starting with an underscore. However these are not allowed as fieldnames in flow.record, example exception:

`flow.record.exceptions.RecordDescriptorError: Field '_5c7566656666_version' is an invalid or reserved field name.`

I fixed it with:

```suggestion
            key = _clean_key(key)

            # Fields that start with an underscore are reserved, add a prefix.
            key = f"n{key}" if key.startswith("_") else key
```

Reply:
Ignore this commit suggestion, I now see that `clean_key` actually converts the the key to hex representation for non ascii bytes and the hex bytes was the representation for the BOM header which I fixed in #272 . 

Adding docstrings and unit tests would be nice.